CONTRIBUTING.md

```diff
@@ -1,35 +1,208 @@
+
+## Before submitting a bug report
+
+-   **DO SOME BASIC TROUBLESHOOTING**:
+    -   restart qBittorrent.
+    -   restart your PC.
+    -   update your OS (e.g. Windows updates).
+    -   update your network card drivers.
+    -   fully reinstall qBittorrent.
+    -   etc...
+-   Make sure the problem is not caused by anti-virus or other program messing with your files.
+-   Check if you can reproduce the problem in the latest version of qBittorrent.
+-   **Check [forum](https://qbforums.shiki.hu/) and [wiki](https://github.com/qbittorrent/qBittorrent/wiki).** You might be able to find the cause of the problem and fix things yourself.
+-   **Check if the issue exists already in the issue tracker.**
+    -   If it does and the issue is still open, add a comment to the existing issue instead of opening a new one.
+    -   If you find a Closed issue that seems like it is the same thing that you're experiencing, open a new issue and include a link to the original issue in the body of your new one.
+-   If the issue is with the search functionality:
+    -   **MAKE SURE YOU HAVE [`python`](https://www.python.org/) INSTALLED CORRECTLY (remember the search functionality requires a working python installation).**
+    -   make sure it is in fact a problem with the search functionality itself, and not a problem with the plugins. If something does not work properly with the search functionality, the first step is to rule out search plugin-related issues.
+        -   For search plugin issues, report on the respective search plugin support page, or at [qbittorrent/search-plugins](https://github.com/qbittorrent/search-plugins).
+
+## Steps to ensure a good bug report
+
+**Follow these guidelines** in order to provide as much useful information as possible right away.
+Otherwise, we've noticed that a lot of your time (and the developers') gets thrown away on exchanging back and forth to get this information.
+
+*   Use a **clear and descriptive title** for the issue to identify the problem.
+
+*   Post only **one specific issue per submission.**
+
+*   **Fill out the issue template properly.**
+
+-   **Make sure you are using qBittorrent on a supported platform.** Do not submit issues which can only be reproduced on beta/unsupported releases of supported operating systems (e.g. Windows 10 Insider, Ubuntu 12.04 LTS, etc).
+These are unstable/unsupported platforms, and in all likelihood, whatever the issue is, it is not related to qBittorrent.
+
+*   **Specify the OS you're using, its version and architecture.**
+    *   Examples: Windows 8.1 32-bit, Linux Mint 17.1 64-bit, Windows 10 Fall creators Update 64-bit, etc.
+
+
+*   **Report only if you can reproduce the issue with an official stable release, a beta release, or with the most recent upstream changes (in this last case specify the specific commit you are on).** (beta testing is encouraged :smile:).
```

Comment:
reproduce requirement is quite strong, some bugs isn't easy to reproduce.
or maybe say: Report only if you run into the issue with an official stable release, a beta release... 

AFAIK there were some bugs in the PortableApps.com version and we don't provide support for that.

Reply:
@Chocobo1 
> reproduce requirement is quite strong, some bugs isn't easy to reproduce.

But if upgrading to the latest version fixes the bug, then it saves us and the user all the hassle.
I think the way the sentence is worded now has a similar meaning to what you propose:
`Upgrade to latest version; Problem still occurs ? file and issue : don't file an issue`.

I didn't know about the PortableApps.com builds. I will add a clause for that.

Reply:
>But if upgrading to the latest version fixes the bug, then it saves us and the user all the hassle.

Right. But I don't see what is related to what I said.
Just to be clear, the "reproduce" here means "able to trigger the bug repetitively" and this shouldn't be a strong requirement to file an issue (as some bugs ain't easy to trigger).

>I didn't know about the PortableApps.com builds. I will add a clause for that.

I think you shouldn't put a link to it (to avoid encouragement), just say we don't provide support out of official releases (on Windows).

Reply:
@Chocobo1 
> Right. But I don't see what is related to what I said.
Just to be clear, the "reproduce" here means "able to trigger the bug repetitively" and this shouldn't be a strong requirement to file an issue (as some bugs ain't easy to trigger).

I understand, will change.

> I think you shouldn't put a link to it (to avoid encouragement), just say we don't provide support out of official releases (on Windows).

Ok